src/button-group/__tests__/button-group-dev.test.tsx

```diff
@@ -6,17 +6,21 @@ import { fireEvent } from '@testing-library/react';
 import { warnOnce } from '@cloudscape-design/component-toolkit/internal';
 
 import { ButtonGroupProps } from '../../../lib/components/button-group';
+import useBaseComponent from '../../../lib/components/internal/hooks/use-base-component';
 import { renderButtonGroup } from './common';
 
 import buttonStyles from '../../../lib/components/button/styles.css.js';
 
+jest.mock('../../../lib/components/internal/hooks/use-base-component', () => jest.fn().mockReturnValue({}));
+
 jest.mock('@cloudscape-design/component-toolkit/internal', () => ({
   ...jest.requireActual('@cloudscape-design/component-toolkit/internal'),
   warnOnce: jest.fn(),
 }));
 
 afterEach(() => {
   (warnOnce as jest.Mock).mockReset();
+  (useBaseComponent as jest.Mock).mockReset();
```

Comment:
Mocking React hooks is an anti pattern, it caused problems in the past, and we do not recommend any teams to do, so we also should not do it in our code.

If you want to test this, do it differently, not via mocking the hook

Reply:
I replaced it with a spy instead so that the original hook is called.

Reply:
The original hook is not called

Reply:
Oh, indeed - apologies! I forgot to remove the old mock after my testing: https://github.com/cloudscape-design/components/pull/3180/commits/9ba450536aed1308b5c03fe67ffb77437456cde9

I did test it by adding the logs to the original hook, so not it is certainly called for each test case.